ieee802154: ieee802154_cc13xx_cc26xx_subg: Add cc1352p7 support

Ayush1325 opened the pull request:
- Add support cc1352p7 used by beagleconnect_freedom

Tested echo_client and echo_server samples on beagleconnect freedom with some dt overlays (since it is a multi-interface device).

Ayush1325 committed:
ieee802154: ieee802154_cc13xx_cc26xx_subg: Add cc1352p7 support

- Add support cc1352p7 used by beagleconnect_freedom
- Since this is a multi interface device, auto config does not work.

Signed-off-by: Ayush Singh <ayushdevel1325@gmail.com>

pdgendt commented on drivers/ieee802154/ieee802154_cc13xx_cc26xx_subg.c:
```diff
@@ -82,7 +82,7 @@ static uint32_t ieee802154_cc13xx_overrides_sub_ghz[] = {
 
 /* Radio values for CC13X2P */
-#elif defined(CONFIG_SOC_CC1352P)
+#elif defined(CONFIG_SOC_CC1352P) | defined(CONFIG_SOC_CC1352P7)
```
```suggestion
#elif defined(CONFIG_SOC_CC1352P) || defined(CONFIG_SOC_CC1352P7)
```

jadonk committed:
soc: arm: ti_simplelink: cc13x2x7_cc26x2x7: add caparray delta config

- Make caparray delta a Kconfig variable
- Set caparray delta for beagle_bcf at beagleconnect_freedom_defconfig

Signed-off-by: Ayush Singh <ayushdevel1325@gmail.com>
Signed-off-by: Jason Kridner <jkridner@beagleboard.org>

vaishnavachath commented on boards/arm/beagle_bcf/beagleconnect_freedom_defconfig:
```diff
@@ -25,3 +25,6 @@ CONFIG_HW_STACK_PROTECTION=y
 CONFIG_UART_CONSOLE=y
+
+# Adjust for oscillator capacitors
+CONFIG_CC13X2_CC26X2_XOSC_CAPARRAY_DELTA=0x02
```
Your commit message does not mention about touching this file/ please explain in commit message.

vaishnavachath commented on soc/arm/ti_simplelink/cc13x2x7_cc26x2x7/Kconfig.soc:
```diff
@@ -65,4 +65,11 @@ config CC13X2_CC26X2_BOOTLOADER_BACKDOOR_LEVEL
+	default 0xD5
+	help
+	  Enable a specific cap array tunning delta.
+
```
Changes looks same as https://github.com/zephyrproject-rtos/zephyr/pull/58877/commits/f0012322701cbec050447859c386b66b293bc49e , in such cases it would be good to retain the authorship of the original author and add your signed-off-by:
